File: MAP/Interpreter C#/Interpreter/VarExp.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Interpreter
{
    namespace Model
    {
        class VarExp:Exp
        {
            string id;

            public VarExp(string id)
            {
                this.id = id;
            }

            public override int eval(MyIDictionary<string, int> tbl)
            {
                return tbl.lookup(this.id);
            }

            public override string ToString()
            {
                return this.id;
            }
        }
    }
}

File: Interpretor2/src/Model/VarExp.java
package Model;

public class VarExp extends Exp {

	String id;
	
	
	public VarExp(String i)
	{
		this.id=i;
		
	}
	
	@Override
	public int eval(MyIDictionary<String, Integer> tbl,MyIDictionary<Integer,Integer> heap) throws Exception {
		return tbl.lookup(id);
	}
	@Override
	public String toString(){
		return id;
	}

}
